src/main.rs

```diff
@@ -164,13 +166,21 @@ fn main() -> anyhow::Result<()> {
     };
 
     let toolchains = match &criterion {
-        Criterion::Installed => Some(vec![]),
-        Criterion::Toolchains(vec) => Some(vec).cloned(),
+        Criterion::Installed => Some(None),
+        Criterion::Toolchains(vec) => Some(Some(vec.clone())),
         _ => None,
     };
     if let Some(toolchains) = toolchains {
+        let hashed_toolchains = match hash_toolchains(toolchains.as_ref()) {
+            Ok(toolchains) => toolchains,
+            Err(err) => {
+                error!("{:?}", err.context("Failed to load toolchains."));
+                return Ok(());
+            }
+        };
```

Comment:
```suggestion
        let hashed_toolchains = hash_toolchains(toolchains.as_ref()).context("failed to load toolchains")?;
```

Reply:
I did this and force-pushed, it broke tests again https://github.com/holmgr/cargo-sweep/actions/runs/4090488399/jobs/7053936897, so I reverted it back to the previous, with `match`.

Reply:
It looks like we're inconsistent in several places about whether we use `error!` or return an error from main - I think we should be doing the latter in all cases, but it doesn't need to block this PR.

```
> rg 'error!' -A2
src/main.rs
263:                    Err(e) => error!(
264-                        "{:?}",
265-                        e.context(format!("Failed to clean {project_path:?}"))
--
293:                    Err(e) => error!("Failed to clean {:?}: {:?}", project_path, e),
294-                };
295-            }
--
321:                    Err(e) => error!("Failed to clean {:?}: {:?}", project_path, e),
```

If you do make that consistent at some point, please update the test instead of reverting the change - it needs to use `stderr` instead of `stdout`.

Reply:
I created #94 to address it, and I'd prefer not solving this inconsistency in this PR (this inconsistency is actually very consistent for not returning an Error, so it's a change worth of another PR).